python/cuml/common/import_utils.py

```diff
@@ -90,12 +90,15 @@ def check_min_cupy_version(version):
         return False
 
 
-def has_scipy():
+def has_scipy(raise_if_unavailable=False):
     try:
         import scipy   # NOQA
         return True
     except ImportError:
-        return False
+        if raise_if_unavailable:
+            return False
+        else:
+            raise ImportError("Scipy is not available.")
```

Comment:
This is the opposite way it should be no? I.e. it is raising if `raise_if_unavailable` is False

Reply:
That was intended to be `if not raise_if_unavailable`. Good catch!